stripe/src/main/java/com/stripe/android/view/StripeEditText.kt

```diff
@@ -89,12 +89,19 @@ open class StripeEditText @JvmOverloads constructor(
         maxLines = 1
         listenForTextChanges()
         listenForDeleteEmpty()
-        determineDefaultErrorColor()
         cachedColorStateList = textColors
+        determineDefaultErrorColor()
     }
 
     protected open val accessibilityText: String? = null
 
+    override fun setTextColor(colors: ColorStateList?) {
+        super.setTextColor(colors)
+
+        // This will only use textColors and not colors because textColor is never null
+        cachedColorStateList = textColors
+    }
```

Comment:
Can we add a new method `setDefaultTextColor`?

Reply:
We could but user's could still call setTextColor?  What are the reservations on overriding this method?

Some things I like about this approach is that it works as expected they set the text color, and when there is an error we set it to the error color, and then will return it back to this when done.  They don't have to know about the special stuff we are doing.

It is just a function rename, so I can do it either way.